### PR TITLE
Add error markers

### PR DESCRIPTION
**Pull request by shaunlebron**
Errors are now marked with `.parinfer-error-marker`.  These error markers are cleared in the parent expression hack range when text is processed.  If there is still an error, the marker is created again.  Some enhancements can be made, but it works.

atom-parinfer-errors

### CODE REVIEW

File: src-cljs/atom_parinfer/core.cljs [hunk 1]
```diff
@@ -273,6 +273,41 @@
...
+  (let [markers (ocall js-editor "findMarkers")]
+                  ; (js-obj "startBufferRow" start-row
+                  ;         "endBufferRow" end-row))]
+    (js/console.log "markers" markers)
+    markers))
+
+(defn clear-previous-error-markers
+  [js-editor start-row end-row js-error]
+  (doseq [marker (get-previous-error-markers js-editor start-row end-row)]
+    ; (ocall marker "destroy"))
```
Comment (shaunlebron) [inline src-cljs/atom_parinfer/core.cljs]: not destroying them yet since I'm not sure how to determine which markers _we_ created

**Comment by shaunlebron** [comment]
`$('.parinfer-error-marker')` in the console turns up null, so the marker decoration is not being created or is being removed by something

File: src-cljs/atom_parinfer/core.cljs [hunk 2]
```diff
@@ -273,6 +273,42 @@
 
 (add-watch *editor-states :toggle-status-classes toggle-status-classes!)
 
+;;------------------------------------------------------------------------------
+;; Error Markers
+;;------------------------------------------------------------------------------
+
+(def error-marker-class "parinfer-error-marker")
```
Comment (oakmac) [inline src-cljs/atom_parinfer/core.cljs]: Should this class be hashed like the other classnames in this extension?

File: src-cljs/atom_parinfer/core.cljs [hunk 3]
```diff
@@ -273,6 +273,42 @@
...
+(defn get-error-markers
+  [js-editor start-row end-row]
+  (->> (ocall js-editor "findMarkers")
+       (filter #(@error-marker-ids (oget % "id")))
+       (filter #(marker-inside? % start-row end-row))))
+
+(defn clear-error-markers
+  [js-editor start-row end-row js-error]
+  (doseq [marker (get-error-markers js-editor start-row end-row)]
+    (swap! error-marker-ids disj (oget marker "id"))
```
Comment (oakmac) [inline src-cljs/atom_parinfer/core.cljs]: Would prefer to not see `swap!` inside a `doseq`

Can you make `swap!` a single operation here?

File: src-cljs/atom_parinfer/core.cljs (hunk 2 above)
Comment (shaunlebron) [inline src-cljs/atom_parinfer/core.cljs]: sorry, I've been trying to match your idioms in this repo, but I think this is the first user-customizable class and I thought a hash might be unnecessary here and maybe jarring to users.

**Comment by shaunlebron** [comment]
made the requested swap into a single operation: https://github.com/oakmac/atom-parinfer/pull/84/commits/885f070e731bdd73f7d5be0b33a7aee805b04878

I used [`run!`](http://cljs.github.io/api/cljs.core/runBANG) to destroy all the markers; hadn't used that one before but I think I like it.

**Comment by shaunlebron** [comment]
warning colors now

screen shot 2017-08-04 at 2 19 12 pm

**Comment by oakmac** [comment]
Does the CSS require the `!important` to work?

**Comment by shaunlebron** [comment]
i thought it was, but it's not.  confirmed it

**Pull request merged**